pkg/cdi/container-edits.go

```diff
@@ -113,6 +113,16 @@ func (e *ContainerEdits) Apply(spec *oci.Spec) error {
 		}
 	}
 
+	if e.NetDevices != nil {
+		// specgen is currently missing functionality to set Linux NetDevices,
+		// so we use a locally rolled function for now.
+		for _, dev := range e.NetDevices {
+			specgenAddLinuxNetDevice(&specgen, dev.HostIf, &oci.LinuxNetDevice{
```

Comment:
For device nodes we have a private `toOCI` method that we use. Does it make sense to implement one for `NetDevice` as well?

Reply:
Yes, definitely. I'll update to follow the same pattern.

Reply:
Added toOCI() and also split out single device validation to a wrapped Validate() to be more in line with the rest.